src/Symfony/Component/Translation/Loader/ArrayLoader.php

```diff
@@ -46,10 +46,12 @@ private function flatten(array $messages): array
         foreach ($messages as $key => $value) {
             if (\is_array($value)) {
                 foreach ($this->flatten($value) as $k => $v) {
-                    $result[$key.'.'.$k] = $v;
+                    if (null !== $v) {
+                        $result[$key.'.'.$k] = (string) $v;
+                    }
                 }
-            } else {
-                $result[$key] = $value;
+            } elseif (null !== $value) {
+                $result[$key] = (string) $value;
```

Comment:
I think we should remove the cast to string, so that stringable objects aren't resolved eagerly, if that makes sense

Reply:
I don't think that `MessageCatalogue` expects non-string input - in 6.x line there are already types that would force cast to string in `get()` and `set()`. If performance is concern, then cache should be used, and stringable objects here would only be a problem in that case.

Reply:
Still: your PR is motivated by nulls, not by non-stringable. Since this is 5.4, we'd better not change the behavior unless we have a compelling reason to.